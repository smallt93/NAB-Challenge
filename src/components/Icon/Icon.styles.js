import styled from 'styled-components';

export const IconWrapper = styled.div`
  position: relative;

  ${({ theme, hover }) =>
    hover &&
    `
        &:hover {
            background-color: ${theme.colors.lightenGrey};

            > div {
                visibility: visible;
            }
        }
    `}
`;

export const IconItem = styled.div`
  background-image: ${({ iconUrl }) => `url(${iconUrl})`};
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 100%;
  width: ${(props) => props.iconSize}rem;
  height: ${(props) => props.iconSize}rem;
`;