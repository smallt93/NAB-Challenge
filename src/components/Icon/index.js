import React from 'react';
import ToolTip from 'components/ToolTip';
import {
  IconWrapper,
  IconItem,
} from './Icon.styles';

const IconComponent = ({
  text = '',
  iconUrl = '',
  iconSize = 1,
  showToolTip = false,
}) => {
  return (
    <IconWrapper hover={showToolTip}>
      <IconItem iconUrl={iconUrl} iconSize={iconSize} />

      {showToolTip && <ToolTip title={text} align='center' />}
    </IconWrapper>
  );
};

export default IconComponent;
