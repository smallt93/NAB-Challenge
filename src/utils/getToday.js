const weekDay = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];

export const getDayOfWeek = (date) => {
  const getDay = String(date.getDate()).padStart(2, '0');
  return `${weekDay[date.getDay()]} ${getDay}`;
};