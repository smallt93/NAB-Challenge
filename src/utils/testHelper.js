import React from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'stylesheets/defaultTheme.styles';


export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

export function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}
