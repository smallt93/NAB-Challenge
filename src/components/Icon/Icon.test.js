import React from 'react';
import { mountWithTheme as mount } from 'utils/testHelper';
import Icon from 'components/Icon';

describe('render Icon component', () => {
  let Component;

  const mockProps = {
    text: '',
    iconUrl: '',
    iconSize: 1,
    showToolTip: false,
  }
  
  beforeEach(() => {
    Component = mount(<Icon {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    expect(Component.length).toBe(1);
  });

  it('should match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });
});
