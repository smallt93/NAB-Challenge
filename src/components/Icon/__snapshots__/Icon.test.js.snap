// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Icon component should match snapshot 1`] = `ReactWrapper {}`;
